### PR TITLE
mediatek: add support for MT8183 SoC

### PR DESCRIPTION
**Pull request by Fabo**
Add OP-TEE support for MT8183 SoC.

### CODE REVIEW

File: core/arch/arm/plat-mediatek/main.c [hunk 1]
```diff
@@ -15,6 +17,37 @@ register_phys_mem_pgdir(MEM_AREA_IO_NSEC,
 
 static struct serial8250_uart_data console_data;
 
+#ifdef CFG_GIC
+static struct gic_data gic_data;
+
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE + GICD_OFFSET,
+	CORE_MMU_PGDIR_SIZE);
```
Comment (etienne-lms) [inline core/arch/arm/plat-mediatek/main.c:24]: indent to the parenthesis.

File: core/arch/arm/plat-mediatek/main.c [hunk 2]
```diff
@@ -15,6 +17,37 @@ register_phys_mem_pgdir(MEM_AREA_IO_NSEC,
...
+register_phys_mem_pgdir(MEM_AREA_IO_SEC, GIC_BASE + GICC_OFFSET,
+	CORE_MMU_PGDIR_SIZE);
+
+void main_init_gic(void)
+{
+	vaddr_t gicc_base;
+	vaddr_t gicd_base;
```
Comment (etienne-lms) [inline core/arch/arm/plat-mediatek/main.c:31]: `= 0;` or
```c
	vaddr_t gicc_base = (vaddr_t)phys_to_virt(GIC_BASE + GICC_OFFSET,
						  MEM_AREA_IO_SEC);
```

File: core/arch/arm/plat-mediatek/platform_config.h [hunk 3]
```diff
@@ -49,6 +49,20 @@
 #define CONSOLE_BAUDRATE	921600
 #define CONSOLE_UART_CLK_IN_HZ	26000000
 
+#elif defined(PLATFORM_FLAVOR_mt8183)
+
+#define GIC_BASE		0x0C000000
+#define GICC_OFFSET		0x400000
+#define GICD_OFFSET		0x00000
```
Comment (etienne-lms) [inline core/arch/arm/plat-mediatek/platform_config.h:56]: prefer `0` or `0x0`.

**Review by etienne-lms — COMMENTED** [review]
`Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with minor coding style comments? Note also minor inconsistency in the 2 commit log header lines: "mediatek: ..." or "plat-mediatek: ..."? nitpicking.

File: core/arch/arm/plat-mediatek/platform_config.h (hunk 3 above)
Comment (Fabo) [reply core/arch/arm/plat-mediatek/platform_config.h]: Fixed

File: core/arch/arm/plat-mediatek/main.c (hunk 2 above)
Comment (Fabo) [reply core/arch/arm/plat-mediatek/main.c]: Fixed

File: core/arch/arm/plat-mediatek/main.c (hunk 1 above)
Comment (Fabo) [reply core/arch/arm/plat-mediatek/main.c]: Fixed

**Comment by Fabo** [comment]
> `Acked-by: Etienne Carriere <etienne.carriere@linaro.org>` with minor coding style comments? Note also minor inconsistency in the 2 commit log header lines: "mediatek: ..." or "plat-mediatek: ..."? nitpicking.

Everything should be fixed in 62ab485

**Comment by jforissier** [comment]
@Fabo thanks!

**Pull request merged**